ocaml/libs/http-lib/server_io.ml

```diff
@@ -26,15 +26,83 @@ type handler = {
   ; lock: Semaphore.Counting.t
 }
 
+module WorkerPool = struct
+  type task = unit -> unit
+
+  type t = {
+      queue: task Queue.t
+    ; size: int
+    ; lock: Mutex.t
+    ; cond_non_empty: Condition.t
+  }
+
+  let worker p =
+    let next () =
+      Mutex.lock p.lock ;
+      while Queue.is_empty p.queue do
+        Condition.wait p.cond_non_empty p.lock
+      done ;
+      let task = Queue.pop p.queue in
+      Mutex.unlock p.lock ;
+      ignore (task ())
```

Comment:
`task` already returns `unit`, there is not need for the `ignore`. It is probably best to avoid `ignore`, because it is very easy to give it a partially applied function as argument, which won't result in any build warnings or errors, but then do nothing at runtime.